src/pyodide/internal/workers.py

```diff
@@ -198,23 +215,41 @@ def redirect(url: str, status: HTTPStatus | int = HTTPStatus.FOUND):
 
     @staticmethod
     def json(
-        data: str | dict[str, str],
+        data: str | dict[str, str] | list | set,
         status: HTTPStatus | int = HTTPStatus.OK,
         status_text="",
         headers: Headers = None,
     ):
         options = Response._create_options(status, status_text, headers)
+        js_resp = None
         with _manage_pyproxies() as pyproxies:
             try:
-                return js.Response.json(
-                    to_js(
-                        data, dict_converter=js.Object.fromEntries, pyproxies=pyproxies
-                    ),
-                    **options,
-                )
+                if isinstance(data, list):
```

Comment:
I think if `data` isn't a JsProxy the best think to do might be to use `json.dumps()` to serialize it in python and then use `Response.text()`.